include/LightGBM/feature_group.h

```diff
@@ -181,6 +185,8 @@ class FeatureGroup {
   std::unique_ptr<Bin> bin_data_;
   /*! \brief True if this feature is sparse */
   bool is_sparse_;
+  /*! \brief Threshold for treating a feature as a sparse feature */
+  double sparse_threshold_;
```

Comment:
Currently this member variable is used in Dataset::CreateValid() and Dataset::CopyFeatureMapperFrom(). When copying a Dataset object to a new Dataset object we need to know the sparse_threshold of the old Dataset to reconstruct the FeatureGroup objects.